File: code.py
# 1. program for swap two number using third variable
# def swap(a,b):
#     c = a 
#     a = b 
#     b = c 
#     print(f"after swapping number a = {a} and b = {b} ")

# a = int(input())
# b = int(input())
# swap(a,b)

#2. program for swapping two number without third variable

# def swap(a,b):
#     a = a+b 
#     b = a-b
#     a = a-b
#     print(f"after swap a = {a} and b = {b}")

# a = int(input())
# b = int(input())
# swap(a,b)

# 3. program for swapping two number using ex-or operator
# def swap(a,b):
#     a = a^b 
#     b = a^b
#     a = a^b
#     print(f"after swap a = {a} and b = {b}")

# a = int(input())
# b = int(input())
# swap(a,b)

# 4. program for swapping three nubers without any extra variable

# def swap(a,b,c):
#     a = a+b+c
#     b = a-b-c
#     c = a-b-c
#     a = a-b-c
#     print(f"after swap a = {a} b = {b} and c = {c}")

# a = int(input())
# b = int(input())
# c = int(input())
# swap(a,b,c)


# 5. program for checking number is even or odd
# def checkNum(n):
#     return n%2==0
# n = int(input())
# if(checkNum(n)):
#     print(f"{n} is even")
# else:
#     print(f"{n} is odd")
    
    
# 6. program for calulate factorials' 

# def fact(n):
#     if(n==0):
#         return 1
#     return n*fact(n-1)
# n = int(input())
# ans = fact(n);
# print(f"{ans} is facorail of {n}")

# 7. program for calculate a sum of digit of a number
# def sumdigt(n):
#     sum = 0
#     while(n):
#         b = n%10
#         sum+=b
#         n = n//10
#     return sum
# n = int(input())
# ans = sumdigt(n)
# print(f"{n} sum of digit is {ans} ")

# 8. program for reverse a number
# def reverseNum(n):
#     revese = 0
#     while(n):
#         b = n%10
#         revese = (revese*10)+b
#         n = n//10
#     return revese
# n = int(input())
# ans = reverseNum(n)
# print(f"{n} of reverse number is {ans} ")

#9. program for check number is palindrome or not
# def checkPalindrome(n):
#     revese = 0
#     original = n
#     while(n):
#         b = n%10
#         revese = (revese*10)+b
#         n = n//10
#     return original == revese
# n = int(input())
# if(checkPalindrome(n)):
#     print(f"{n} is palindrome")
# else:
#     print(f"{n} is not a palindrome")

#10. program for check number is strong number or not 
# def fact(n):
#     if(n==0):
#         return 1
#     return n*fact(n-1)

# def CheckStrongnum(n):
#     factdigit = 0
#     original = n
#     while(n):
#         b = n%10
#         factdigit = factdigit+fact(b)
#         n = n//10
#     return original == factdigit
# n = int(input())
# if(CheckStrongnum(n)):
#     print(f"{n} is strong number")
# else:
#     print(f"{n} is not a strong number")

#11 program for check year is leap year or not

# def CheckYear(year):
#     if(year%400==0 or (year%4==0 and year%100!=0)):
#         print(f"{year} is leap year")
#     else:
#         print(f"{year} is normal year")


# year = int(input())
# CheckYear(year)
    
# 12 . program for calculate product of a number

# def productNum(n):
#     product = 1
#     while(n):
#         b = n%10
#         product = product*b 
#         n = n//10
#     return product
# n = int(input())
# ans = productNum(n)
# print(f"{ans} is product of {n} number")

# 13. program for check number is armstrong number or not

# def counter(n):
#     count = 0
#     while(n):
#         b = n%10
#         count = count+1
#         n = n//10
#     return count
# def armStrong(n):
#     original = n
#     count = counter(n)
#     value = 0
#     while(n):
#         b = n%10
#         value = value+pow(b, count)
#         n = n//10
#     return value == original
# n = int(input())
# if(armStrong(n)):
#     print(f"{n} is armstrong ")
# else:
#     print(f"{n} not strong muber")

# 14. program for check number is Disarium number or not

# def counter(n):
#     count = 0
#     while(n):
#         b = n%10
#         count = count+1
#         n = n//10
#     return count
# def DisariumNUm(n):
#     original = n
#     count = counter(n)
#     value = 0
#     while(n):
#         b = n%10
#         value = value+pow(b, count)
#         n = n//10
#         count = count-1
#     return value == original
# n = int(input())
# if(DisariumNUm(n)):
#     print(f"{n} is Disarium ")
# else:
#     print(f"{n} not Disarium muber")

#15. program for check number is SPY number or not

# def SpyNum(n):
#     Sum = 0
#     product = 1
#     while(n):
#         b =n%10
#         Sum = Sum+b 
#         product = product * b
#         n= n//10
#     return Sum == product
# n = int(input())
# if(SpyNum(n)):
#     print(f"{n} is spy number")
# else:
#     print("not a spy number")

# 16 . program for check number is prime or not 

# import math
# def primeNum(n):
#     if(n<=0):
#        return False
#     else:
#         square = math.sqrt(n)
#         i  =2
#         while(i<=square):
#             if(n%i==0):
#                 return False
#             i = i+1
#         return True

# n = int(input())
# if(primeNum(n)):
#     print("prime number")
# else:
#     print("not prime")

        
        
#17. sum of array element 
# cook your dish here
# def sumOfArry(n,arr):
#     sum = 0;
#     for i in range (n):
#         sum = sum+arr[i]
#     return sum;
# n = int(input())
# arr = list(map(int,input().split()))
# ans = sumOfArry(n,arr)
# print(ans)

File: code.cpp
// #include <iostream>
// #include <cmath>
// using namespace std;

// 1. program for swapping two numbers with the help of a third variable
// void swap(int a, int b) {
//     int c;
//     c = a;
//     a = b;
//     b = c;
//     cout << "After swapping a = " << a << " and b = " << b;
// }

// int main() {
//     int a, b;
//     cin >> a >> b;
//     swap(a, b);
//     return 0;
// }

// 2. program for swapping two numbers without a third variable
// void swapWithoutThird(int a, int b) {
//     a = a + b;
//     b = a - b;
//     a = a - b;
//     cout << "After swapping a = " << a << " and b = " << b;
// }

// int main() {
//     int a, b;
//     cin >> a >> b;
//     swapWithoutThird(a, b);
//     return 0;
// }

// 3. program for swapping two numbers by using XOR operator
// void swapUsingXOR(int a, int b) {
//     a = a ^ b;
//     b = a ^ b;
//     a = a ^ b;
//     cout << "After swapping a = " << a << " and b = " << b;
// }

// int main() {
//     int a, b;
//     cin >> a >> b;
//     swapUsingXOR(a, b);
//     return 0;
// }

// 4. program for swapping three numbers without a third variable
// void swapThree(int a, int b, int c) {
//     a = a + b + c;
//     b = a - b - c;
//     c = a - b - c;
//     a = a - b - c;
//     cout << "After swapping a = " << a << ", b = " << b << ", and c = " << c;
// }

// int main() {
//     int a, b, c;
//     cin >> a >> b >> c;
//     swapThree(a, b, c);
//     return 0;
// }

// 5. program for checking if a year is a leap year or not
// void checkYear(int year) {
//     if (year % 400 == 0) {
//         cout << "Year is a leap year" << ;
//     } else if (year % 4 == 0 && year % 100 != 0) {
//         cout << "Leap year";
//     } else {
//         cout << "Not a leap year" ;
//     }
// }

// int main() {
//     int year;
//     cin >> year;
//     checkYear(year);
//     return 0;
// }

// 6. program for sum of digits of a number
// int sumOfDigit(int num) {
//     int b, sum = 0;
//     while (num) {
//         b = num % 10;
//         sum += b;
//         num /= 10;
//     }
//     return sum;
// }

// int main() {
//     int number;
//     cin >> number;
//     cout << "Sum of digits is " << sumOfDigit(number);
//     return 0;
// }

// 7. program for reversing a number
// int reverseNum(int num) {
//     int b, reversedNumber = 0;
//     while (num) {
//         b = num % 10;
//         reversedNumber = (reversedNumber * 10) + b;
//         num /= 10;
//     }
//     return reversedNumber;
// }

// int main() {
//     int number;
//     cin >> number;
//     cout << "Reversed number is " << reverseNum(number);
//     return 0;
// }

// 8. program for checking if a number is a palindrome or not
// bool checkPalindrome(int num) {
//     int b, reverseNum = 0, original = num;
//     while (num) {
//         b = num % 10;
//         reverseNum = (reverseNum * 10) + b;
//         num /= 10;
//     }
//     return reverseNum == original;
// }

// int main() {
//     int number;
//     cin >> number;
//     if (checkPalindrome(number)) {
//         cout << number << " is a palindrome";
//     } else {
//         cout << number << " is not a palindrome";
//     }
//     return 0;
// }

// 9. program for calculating the factorial of a number
// int fact(int num) {
//     if (num == 0) {
//         return 1;
//     }
//     return num * fact(num - 1);
// }

// int main() {
//     int num;
//     cin >> num;
//     cout << fact(num) << " is the factorial of this number " << num ;
//     return 0;
// }

// 10. program for checking if a number is a strong number
// bool checkStrongNum(int n) {
//     int b, value = 0, temp = n;
//     while (n) {
//         b = n % 10;
//         value += fact(b);
//         n /= 10;
//     }
//     return value == temp;
// }

// int main() {
//     int number;
//     cin >> number;
//     if (checkStrongNum(number)) {
//         cout << number << " is a strong number" ;
//     } else {
//         cout << number << " is not a strong number";
//     }
//     return 0;
// }

// 11. program for checking if a number is even or odd
// bool checkNum(int n) {
//     return n % 2 == 0;
// }

// int main() {
//     int n;
//     cin >> n;
//     if (checkNum(n)) {
//         cout << n << " is even";
//     } else {
//         cout << n << " is odd";
//     }
//     return 0;
// }

// 12. program for calculating the product of digits of a number
// int productNum(int n) {
//     int b, value = 1;
//     while (n) {
//         b = n % 10;
//         value *= b;
//         n /= 10;
//     }
//     return value;
// }

// int main() {
//     int n;
//     cin >> n;
//     cout << productNum(n) << " is the product of the digits of " << n;
//     return 0;
// }

// 13. program for checking if a number is a SPY number or not
// bool SpyNum(int n) {
//     int b, product = 1, sum = 0;
//     while (n) {
//         b = n % 10;
//         product *= b;
//         sum += b;
//         n /= 10;
//     }
//     return product == sum;
// }

// int main() {
//     int n;
//     cin >> n;
//     if (SpyNum(n)) {
//         cout << n << " is a SPY number" ;
//     } else {
//         cout << n << " is not a SPY number";
//     }
//     return 0;
// }

// 14. program for checking if a number is an Armstrong number
// int CalcuCount(int n) {
//     int b, count = 0;
//     while (n) {
//         b = n % 10;
//         count++;
//         n /= 10;
//     }
//     return count;
// }

// bool armstrongNum(int n) {
//     int b, value = 0, count = CalcuCount(n), a = n;
//     while (n) {
//         b = n % 10;
//         value += pow(b, count);
//         n /= 10;
//     }
//     return value == a;
// }

// int main() {
//     int n;
//     cin >> n;
//     if (armstrongNum(n)) {
//         cout << n << " is an Armstrong number";
//     } else {
//         cout << n << " is not an Armstrong number";
//     }
//     return 0;
// }

// 15. program for checking if a number is a Disarium number
// bool DisariumNum(int n) {
//     int b, value = 0, count = CalcuCount(n), a = n;
//     while (n) {
//         b = n % 10;
//         value += pow(b, count);
//         n /= 10;
//         count--;
//     }
//     return value == a;
// }

// int main() {
//     int n;
//     cin >> n;
//     if (DisariumNum(n)) {
//         cout << n << " is a Disarium number" ;
//     } else {
//         cout << n << " is not a Disarium number" ;
//     }
//     return 0;
// }

 // 16. Program to check if a number is a Sunny number
// #include <iostream>
// #include <cmath>
// using namespace std;

// int main() {
//     int n, sunny;
//     float ans;
//     cin >> n;
//     sunny = n + 1;
//     ans = sqrt(sunny);
//     if (ans == static_cast<int>(ans)) {
//         cout << "Sunny number";
//     } else {
//         cout << "Not a Sunny number";
//     }
//     return 0;
// }

 // 17. Program to check if a number is a prime number
// #include <iostream>
// #include <cmath>
// using namespace std;

// bool CheckPrime(int value) {
//     if (value <= 1) {
//         return false;
//     }
//     int square = sqrt(value);
//     for (int i = 2; i <= square; i++) {
//         if (value % i == 0) {
//             return false;
//         }
//     }
//     return true;
// }

// int main() {
//     int n;
//     cin >> n;
//     if (CheckPrime(n)) {
//         cout << n << " is a prime number";
//     } else {
//         cout << n << " is not a prime number";
//     }
//     return 0;
// }

// 18. Program to print Fibonacci series
// #include <iostream>
// using namespace std;

// void fibonacciSeries(int n) {
//     if (n == 0) {
//         cout << "0\n";
//     } else if (n == 1) {
//         cout << "0 1\n";
//     } else {
//         int a = 0, b = 1;
//         cout << "0 1 ";
//         n -= 2;
//         while (n) {
//             int c = a + b;
//             cout << c << " ";
//             a = b;
//             b = c;
//             n--;
//         }
//         cout << endl;
//     }
// }

// int main() {
//     int n;
//     cin >> n;
//     fibonacciSeries(n);
//     return 0;
// }

 // 19. Program to check if a number is in the Fibonacci series
// #include <iostream>
// #include <cmath>
// using namespace std;

// bool checkNum(int n) {
//     double a = sqrt(5 * n * n + 4);
//     double b = sqrt(5 * n * n - 4);
//     return a == static_cast<int>(a) || b == static_cast<int>(b);
// }

// int main() {
//     int n;
//     cin >> n;
//     if (checkNum(n)) {
//         cout << n << " is present in the Fibonacci series.";
//     } else {
//         cout << n << " is not present in the Fibonacci series.";
//     }
//     return 0;
// }

// 20. Program to calculate the sum of array elements
// #include <iostream>
// #include <vector>
// using namespace std;

// int sumOfArray(int n, const vector<int>& arr) {
//     int sum = 0;
//     for (int i = 0; i < n; i++) {
//         sum += arr[i];
//     }
//     return sum;
// }

// int main() {
//     int n, ans;
//     cin >> n;
//     vector<int> arr(n);
//     for (int i = 0; i < n; i++) {
//         cin >> arr[i];
//     }
//     ans = sumOfArray(n, arr);
//     cout << "Sum of array elements is " << ans;
//     return 0;
// }
